src/UiPath.Workflow.Runtime/Validation/ActivityValidationServices.cs

```diff
@@ -453,9 +454,28 @@ internal ValidationResults InternalValidate()
                 ActivityUtilities.CacheRootMetadata(_rootToValidate, _environment, _options, new ActivityUtilities.ProcessActivityCallback(ValidateElement), ref _errors);
             }
 
+            var extension = GetValidationExtension(_environment);
```

Comment:
As expressions are the concern of UiPath.Workflow project, expression validation should also be implemented there.